apps: Set Thanos as Default and Deprecated support for InfluxDB (#849)
elastisys/compliantkubernetes-apps: WIP-CHANGELOG.md

        By default, an ingressClass called `nginx` will be available in the cluster.
        Ingress-nginx will still handle ingresses that do not specify an `ingressClassName`, however users are strongly encouraged to update their Ingress Objects and specify `spec.ingressClassName: nginx`.
    - The entire changelog can be found [here](https://github.com/kubernetes/ingress-nginx/blob/main/Changelog.md).

> - InfluxDb is deprecated and Thanos is now enabled by default.

Reviewer: ```suggestion
- InfluxDB is deprecated and Thanos is now enabled by default.
```

>   If you have the nfs-provisioner or the local-pv-provisioner installed, they will be left untouched when upgrading.
>   You are responsible for managing them and/or removing them, and any unsused storageClasses.
> 
> - InfluxDB is deprecated and Thanos is now enabled by default.
>    - With the removal of InfluxDB, the backups and buckets can eventually be removed.

Reviewer 2: ```suggestion
  If you have the nfs-provisioner or the local-pv-provisioner installed, they will be left untouched when upgrading.
  You are responsible for managing them and/or removing them, and any unsused storageClasses.
- InfluxDB is removed and Thanos is now enabled by default.
   - With the removal of InfluxDB, the backups and buckets can eventually be removed.
 - Running without object storage is no longer supported since it is required for Thanos.
   - The dev flavor is now updated to use s3 by default   
```

 - Upgrade velero helm chart to `v2.27.3`, which also upgrades velero to `v1.7.1`.
 - Upgrade prometheus-elasticsearch-exporter helm chart to v4.11.0 and prometheus-elasticsearch-exporter itself to v1.3.0
 - Exposed options for starboard-operator to control the number of jobs it generates and to allow for it to be disabled.

>  - Thanos is now enabled by default

Reviewer: ```suggestion
 - Thanos is now enabled by default
 
```

Reviewer 2: This does not look like it is fixed. There is still no empty line between this like and `### Fixed`.

 - Upgrade prometheus-elasticsearch-exporter helm chart to v4.11.0 and prometheus-elasticsearch-exporter itself to v1.3.0
 - Exposed options for starboard-operator to control the number of jobs it generates and to allow for it to be disabled.
 - Added the new OPA policy - disallowed the latest image tag.
 - Thanos is now enabled by default
### Fixed

### Added

### Removed
- Removed disabled helm releases from the application helmfile

> - Deprecated influxDB and dependent helm charts

Reviewer 2: ```suggestion
- Removed influxDB and dependent helm charts
```

 - Added the new OPA policy - disallowed the latest image tag.
 - Moved `user.alertmanager.group_by` to `prometheus.alertmanagerSpec.groupBy` in `sc-config.yaml`
 - Moved `user.grafana.userGroups` to `user.grafana.oidc.userGroups` in `sc-config.yaml`

>  - Thanos is now enabled by default

Reviewer 3: I think this should be more clear (put in a `Release notes` section in the top of the file)

Author: Done

 - kubeconfig.bash have been edited to work with the new 'secret' structure.
 - memory limit for thanos receiveDistributor and pvc size for thanos receiver
 - Increased cpu requests and limits for kube-state-metrics
 - Thanos is now enabled by default.

>  - Running without object storage is no longer supported since it is required for Thanos.

Reviewer: This feels release notes-worthy to me.

Author: I added it in the release notes

>  - Running without object storage is no longer supported since it is required for Thanos.
>    - The dev flavor is now updated to use s3 by default

Reviewer 2: ```suggestion
```

 - kubeconfig.bash have been edited to work with the new 'secret' structure.
 - memory limit for thanos receiveDistributor and pvc size for thanos receiver
 - Increased cpu requests and limits for kube-state-metrics
 - Thanos is now enabled by default.

> 

Reviewer: ```suggestion
```
